### ISSUE TITLE
Add Environment variable to disable InfluxDB analytics message 

### ISSUE THREAD

**Issue by zwimer** [Homebrew/brew#15719]
### Verification

- [X] This issue's title and/or description do not reference a single formula e.g. `brew install wget`. If they do, open an issue at https://github.com/Homebrew/homebrew-core/issues/new/choose instead.

### Provide a detailed description of the proposed feature

When running `brew update`, if analytics are disabled the following message is printed:
```
==> Homebrew's analytics have entirely moved to our InfluxDB instance in the EU.
We gather less data than before and have destroyed all Google Analytics data:
  https://docs.brew.sh/Analytics
Please reconsider re-enabling analytics to help our volunteer maintainers with:
  brew analytics on
```

The feature suggestion is an environment variable that disables this message if analytics are off. Something like `HOMEBREW_NO_ANALYTICS_MESSAGE`.

### What is the motivation for the feature?

Sometimes users will have analytics off; notifying them every update that they should turn them back on is annoying and not user friendly.

### How will the feature be relevant to at least 90% of Homebrew users?

It will likely be useful to the same users who use `HOMEBREW_NO_ANALYTICS`. Some may opt not to since google analytics has been removed, but others may prefer this option.

### What alternatives to the feature have been considered?

Provide a mechanism within brew itself to edit this config; the environment variable seems simpler and is what is already done for analytics by default via `HOMEBREW_NO_ANALYTICS`.

**Comment by RandomDSdevel** [comment]
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Is setting the boolean value '`homebrew.installfromapimessage`' to '`true`' in '`"$(brew --repository)/.git/config"`' (either by editing the file manually or by running the appropriate '`git config`' command invocation while in that working directory) not working?

**Comment by zwimer** [comment]
It is immediately overwritten upon invocation of `brew update`:
```
$ cat /opt/homebrew/.git/config
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
	autocrlf = false
	fsmonitor = false
	symlinks = true
[remote "origin"]
	url = https://github.com/Homebrew/brew
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[homebrew]
	analyticsmessage = false
	caskanalyticsmessage = false
	donationmessage = false
	untapped = "homebrew/cask;homebrew/core"
	influxanalyticsmessage = false
	analyticsdisabled = true
$ brew update -f
==> Homebrew's analytics have entirely moved to our InfluxDB instance in the EU.
We gather less data than before and have destroyed all Google Analytics data:
  https://docs.brew.sh/Analytics
Please reconsider re-enabling analytics to help our volunteer maintainers with:
  brew analytics on
==> Homebrew is run entirely by unpaid volunteers. Please consider donating:
  https://github.com/Homebrew/brew#donations

Already up-to-date.
$ cat /opt/homebrew/.git/config
[core]
	repositoryformatversion = 0
	filemode = true
	bare = false
	logallrefupdates = true
	ignorecase = true
	precomposeunicode = true
	autocrlf = false
	fsmonitor = false
	symlinks = true
[remote "origin"]
	url = https://github.com/Homebrew/brew
	fetch = +refs/heads/*:refs/remotes/origin/*
[branch "master"]
	remote = origin
	merge = refs/heads/master
[homebrew]
	analyticsmessage = false
	caskanalyticsmessage = false
	donationmessage = true
	untapped = "homebrew/cask;homebrew/core"
	influxanalyticsmessage = true
	analyticsdisabled = true
```

**Comment by zwimer** [comment]
Adding that to the git config does not change anything when I subsequently invoke `brew update -f`. Additionally, changing the values of variables name `influxanalyticsmessage` does not change anything as `brew update -f` immediately overwrites them the next run before printing the message as usual.

**Comment by nandahkrishna** [comment]
The message was intended to be displayed only once, and not after every `brew update`. There was a bug reported in https://github.com/Homebrew/brew/issues/15678 which has since been fixed (https://github.com/Homebrew/brew/pull/15688 and https://github.com/Homebrew/brew/pull/15695), and will be included in the next release (which will be sometime soon). See also: https://github.com/Homebrew/brew/issues/15718.
